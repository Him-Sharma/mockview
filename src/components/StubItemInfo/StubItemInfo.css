.StubItemInfo {
  text-align: left;
  border: 2px solid #09d3ac;
  color: white;
  flex-basis: 30%;
  margin: 20px 20px;
}
