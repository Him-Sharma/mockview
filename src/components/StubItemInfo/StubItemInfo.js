import React from "react";
import "./StubItemInfo.css";

const stubItemInfo = (props) => {
  let stubInformation = <h1>Select a stub to view information here.</h1>;
  if (props.selectedStubItem) {
    stubInformation = (
      <div>
        <h1>Method: {props.selectedStubItem.predicates[0].equals.method}</h1>
        <h1>Path: {props.selectedStubItem.predicates[0].equals.path}</h1>
        <h1>Responses: </h1>
        <p>StatusCode: {props.selectedStubItem.responses[0].is.statusCode}</p>
        <p>body: {props.selectedStubItem.responses[0].is.body}</p>
      </div>
    );
  }
  return (
    <div className="StubItemInfo" data-testid="messageText">
      {stubInformation}
    </div>
  );
};
export default stubItemInfo;
