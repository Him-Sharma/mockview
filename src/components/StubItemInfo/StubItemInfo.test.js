import React from "react";
import { render } from "@testing-library/react";
import StubItemInfo from "./StubItemInfo";

it("renders correctly", () => {
  const { baseElement } = render(<StubItemInfo null />);
  expect(baseElement).toMatchSnapshot();
});

it("should display stubItem information", () => {
  const selectedStubItem = {
    predicates: [
      {
        equals: {
          method: "GET",
          path: "/posts",
        },
      },
    ],
    responses: [
      {
        is: {
          statusCode: "200",
          body: "{}",
        },
      },
    ],
    _links: {
      self: {
        href: "http://test",
      },
    },
  };
  const { baseElement } = render(
    <StubItemInfo selectedStubItem={selectedStubItem} />
  );
  expect(baseElement).toMatchSnapshot();
});
