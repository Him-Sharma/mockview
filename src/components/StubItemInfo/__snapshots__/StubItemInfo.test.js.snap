// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <div
      class="StubItemInfo"
      data-testid="messageText"
    >
      <h1>
        Select a stub to view information here.
      </h1>
    </div>
  </div>
</body>
`;

exports[`should display stubItem information 1`] = `
<body>
  <div>
    <div
      class="StubItemInfo"
      data-testid="messageText"
    >
      <div>
        <h1>
          Method: 
          GET
        </h1>
        <h1>
          Path: 
          /posts
        </h1>
        <h1>
          Responses: 
        </h1>
        <p>
          StatusCode: 
          200
        </p>
        <p>
          body: 
          {}
        </p>
      </div>
    </div>
  </div>
</body>
`;
